### PR TITLE
Investigate potential rst grammar to include as a built-in extension

### LINKED ISSUE #144567: Investigate potential rst grammar to include as a built-in extension
It would be great to have syntax highlighting for rst files built-in to VS Code.

### PR DESCRIPTION
**Pull request by alexr00**
Fixes #144567

### CODE REVIEW

**Comment by brettcannon** [comment]
How big/small of a sample do you want? All of Python's documentation is written in reST and I know it pretty well, so we could get you a sample as big/thorough as you want it to be.

**Comment by alexr00** [comment]
A sample that has a variety of token types is good. For example, this is the C# sample we use:

https://github.com/microsoft/vscode/blob/8de5c819a1e9aea852c5b5911041a653746ff052/extensions/vscode-colorize-tests/test/colorize-fixtures/test.cs#L1-L27

**Comment by brettcannon** [comment]
Is is probably more thorough than necessary, but I figured it's easier to cut it down than leave something out. 😁

Normally the file would have a `.rst` extension, but GH won't let me upload such a file name, hence the `.txt` extension.

[restructuredtext_sample.txt](https://github.com/microsoft/vscode/files/8217804/restructuredtext_sample.txt)

File: extensions/vscode-colorize-tests/test/colorize-fixtures/test.rst [hunk 1]
```diff
@@ -0,0 +1,3 @@
+============
```
[the remark this answers is not in the document]
Comment (alexr00) [reply extensions/vscode-colorize-tests/test/colorize-fixtures/test.rst:1]: I've updated with the suggested sample from @brettcannon.

**Pull request merged**